tools/run-image.toit

```diff
@@ -43,20 +57,179 @@ class SystemImage extends ContainerImage:
   delete -> none:
     unreachable  // Not implemented yet.
 
+class FirmwareServiceProvider extends FirmwareServiceProviderBase:
+  ota-dir-active_/string?
+  ota-dir-inactive_/string?
+
+  constructor --ota-dir-active/string? --ota-dir-inactive/string?:
+    ota-dir-active_ = ota-dir-active
+    ota-dir-inactive_ = ota-dir-inactive
+    super "system/firmware" --major=0 --minor=1
+
+  is-validation-pending -> bool:
+    return false
+
+  is-rollback-possible -> bool:
+    return false
+
+  validate -> bool:
+    throw "UNIMPLEMENTED"
+
+  rollback -> none:
+    throw "UNIMPLEMENTED"
+
+  upgrade -> none:
+    if not ota-dir-inactive_: throw "No OTA directory"
+    // Exit and tell the outer script that an update is available.
+    exit 17
+
+  config-ubjson -> ByteArray:
+    return file.read-content "$ota-dir-active_/$CONFIG-FILE-NAME_"
+
+  config-entry key/string -> any:
+    return (ubjson.decode config-ubjson)[key]
+
+  content -> ByteArray:
+    if not ota-dir-active_: throw "No OTA directory"
+    return file.read-content "$ota-dir-active_/$BITS-FILE-NAME_"
+
+  uri -> string?:
+    return null
+
+  firmware-writer-open client/int from/int to/int -> FirmwareWriter:
+    if not ota-dir-inactive_: throw "No OTA directory"
+    return FirmwareWriter_ this client from to --ota-dir-inactive=ota-dir-inactive_
+
+class FirmwareWriter_ extends services.ServiceResource implements FirmwareWriter:
+  static image/ByteArray := #[]
+  view_/ByteArray? := null
+  cursor_/int := 0
+  ota-dir-inactive_/string
+
+  constructor provider/FirmwareServiceProvider client/int from/int to/int --ota-dir-inactive/string:
+    ota-dir-inactive_ = ota-dir-inactive
+    if to > image.size: image = image + (ByteArray to - image.size: random 0x100)
```

Comment:
Add a comment to explain why you do this random padding?

Reply:
done.